docker/build_matrix.yaml

```diff
@@ -1,73 +1,84 @@
 # This file is automatically generated by generate_build_matrix.py. DO NOT EDIT!
 - AWS_OFI_NCCL_VERSION: ''
-  BASE_IMAGE: nvidia/cuda:11.8.0-cudnn8-devel-ubuntu20.04
-  CUDA_VERSION: 11.8.0
-  IMAGE_NAME: torch-2-0-1-cu118
+  BASE_IMAGE: nvidia/cuda:12.1.0-cudnn8-devel-ubuntu20.04
+  CUDA_VERSION: 12.1.0
+  IMAGE_NAME: torch-2-1-0-cu121
   MOFED_VERSION: 5.5-1.0.3.2
   PYTHON_VERSION: '3.10'
   PYTORCH_NIGHTLY_URL: ''
   PYTORCH_NIGHTLY_VERSION: ''
-  PYTORCH_VERSION: 2.0.1
+  PYTORCH_VERSION: 2.1.0
   TAGS:
-  - mosaicml/pytorch:2.0.1_cu118-python3.10-ubuntu20.04
+  - mosaicml/pytorch:2.1.0_cu121-python3.10-ubuntu20.04
+  - mosaicml/pytorch:latest
```

Comment:
can we bump latest in a separate PR after we've had a little time to make sure nothing is horribly broken?